### ISSUE TITLE
MQTT failed to connect: Connection refused: Not authorized

### ISSUE THREAD

**Issue by rjmediallp** [Koenkk/zigbee2mqtt#11148]
### What happened?

I updated zigbee2mqtt today (from 1.17 I think to 1.22.2-1) and having restarted, I get an error message "MQTT failed to connect: Connection refused: Not authorized". 

### What did you expect to happen?

MQTT connects and plugin starts up

### How to reproduce it (minimal and precise)

Deleted zigbee2mqtt from home assistant. Added the new repository address and installed v1.22.2-1. Added my Sonof USB adaptor and restarted the plugin.

Home assistant core: 2021.12.10
Supervisor: 2021-12.2
Mosquitto broker v6.0.1

Zigbee2mqtt config:
```
data_path: /config/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt,
  user: mqtt-user,
  password: ********
  server: http://10.1.1.48:1883
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_7ac0421bb0c9eb1188d98b4f1d69213e-if00-port0
advanced:
  log_level: warn
  pan_id: 6754
  channel: 25
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental: {}
availability: false
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: '-d -d'
  log: false
```
I should add that I've tried different servers (with and without). The default previous which worked was
"server": "mqtt://core-mosquitto:1883",

### Zigbee2MQTT version

1.22.2-1

### Adapter firmware version

unknown - bad gateway

### Adapter

ITead Sonoff Zigbee 3.0 USB Dongle Plus

### Debug log

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] socat.sh: executing... 
[12:51:31] INFO: Socat not enabled, marking service as down
[cont-init.d] socat.sh: exited 0.
[cont-init.d] zigbee2mqtt.sh: executing... 
[12:51:33] INFO: MQTT available, fetching server detail ...
[12:51:34] INFO: Previous config file found, checking backup
[12:51:34] INFO: Creating backup config in '/config/zigbee2mqtt/.configuration.yaml.bk'
[12:51:35] INFO: Adjusting Zigbee2mqtt core yaml config with add-on quirks ...
[cont-init.d] zigbee2mqtt.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[12:51:38] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.22.2 start
> node index.js
Zigbee2MQTT:error 2022-01-31 12:51:56: MQTT failed to connect: Connection refused: Not authorized
Zigbee2MQTT:error 2022-01-31 12:51:56: Exiting...
```

**Comment by Koenkk** [comment]
`MQTT failed to connect: Connection refused: Not authorized` 

your mqtt credential are incorrect.

**Comment by rjmediallp** [comment]
Would there be anything between 1.17 and 1.22 which would break the listening on  the server:1883 or something? It's definitely working as I said, I can see the other MQTT elements reporting back. 

![image](https://user-images.githubusercontent.com/6358761/151848454-8d933226-dcaa-48b5-bda3-e98a59f079a4.png)

**Comment by rjmediallp** [comment]
Update: I kept all user settings and MQTT broker as-is... then removed the Zigbee2Mqtt plugin. I noticed I had a repository from both the z2m website as well as the github one. I removed the website repo and installed 1.22.2-1 from github. I then added the mqtt user which I know connects fine and it's all working again. No idea if that was the problem (having two repos and possibly installing a website version rather than the github?) 
Closing now.

**Comment by henrikrox** [comment]
@rjmediallp I'm having the same issue seems @Koenkk just gave a generalised answer that wasn't very helpful. What do you mean website repo and GitHub repo? can you make a list of what you did to fix this, driving me insane

**Comment by djblogi** [comment]
@henrikrox  in my case i had the config for zigbee 2 mqtt like this 
mqtt:
  base_topic: zigbee2mqtt
  user: mqtt-user
  pass: mqtt-pass
  server: mqtt://core-mosquitto:1883

And changed it to

mqtt:
  base_topic: zigbee2mqtt
  user: "mqtt-user"
  password: "mqtt-pass"
  server: mqtt://core-mosquitto:1883

that worked for me.

**Comment by xoref** [comment]
In case anyone has a similar login issue; 
My system crashed and I had to do a `fsck -f` filesystem repair to get it all back up. Afterwards the only thing that was broken in my setup was the Mosquitto broker. It would not accept the login I had been using for years (not via MQTT Explorer nor Zigbee2MQTT). 

**TL;DR;**
I simply changed the login a little, restarted Mosquitto and it started working again.

**Comment by radek-kafka** [comment]
I have found similar problem within my setup. I have supervised version of Homeassistant on RPI4 and after a reboot I need to manually restart MQTT add-on.. 1st attempt to start failing with "Not authorized ..." message for other messages and also from outside (MQTT Explorer) I could not connect to ..

**Comment by DannyFeliz** [comment]
> I have found similar problem within my setup. I have supervised version of Homeassistant on RPI4 and after a reboot I need to manually restart MQTT add-on.. 1st attempt to start failing with "Not authorized ..." message for other messages and also from outside (MQTT Explorer) I could not connect to ..

Same here, I solved it by restarting Mosquitto

**Comment by radek-kafka** [comment]
> > I have found similar problem within my setup. I have supervised version of Homeassistant on RPI4 and after a reboot I need to manually restart MQTT add-on.. 1st attempt to start failing with "Not authorized ..." message for other messages and also from outside (MQTT Explorer) I could not connect to ..
> 
> Same here, I solved it by restarting Mosquitto

The solution I have found is:

add into configuration.yaml

mqtt:
  broker: "core-mosquitto"
  username: "username"
  password: "password"

Now it survives restart without any additional user interaction needed ...

**Comment by buteo72** [comment]
For me it was that I had the original files in configuration.yaml in \**share**\zigbee2mqtt\ and after the update it looked at \**config**\zigbee2mqtt. In that configuration.yaml file, there was no password.

**Comment by DzgHomeCastle** [comment]
In my case the problem with "not authorized MQTT fail" was in mosquitto configuration.
Everything work after add in mosquitto brocker:

logins:
  - username: mqtt-user
    password: *******

**Comment by Arfitzuy** [comment]
For future future travellers, essentially make your password alphanumeric only (I had a # in mine that likely borked it), and be sure to change it on both ends. Reconfigure the MQTT integration and the Zigbee2mqtt side either in settings>MQTT, or in the configuration.yaml. I was initially struggling because I forgot the MQTT integration side.

**Comment by AlinGoga** [comment]
> @henrikrox in my case i had the config for zigbee 2 mqtt like this mqtt: base_topic: zigbee2mqtt user: mqtt-user pass: mqtt-pass server: mqtt://core-mosquitto:1883
> 
> And changed it to
> 
> mqtt: base_topic: zigbee2mqtt user: "mqtt-user" password: "mqtt-pass" server: mqtt://core-mosquitto:1883
> 
> that worked for me.

Thanks mate! it worked for me too ! :)

**Comment by anotherfatduck** [comment]
> In my case the problem with "not authorized MQTT fail" was in mosquitto configuration. Everything work after add in mosquitto brocker:
> 
> ```
> loggins:
>    - username: mqtt-user
>      password: Anim@lsH0m3
> ```

this sorted out my issue as well.
thank you.

**Comment by milkpirate** [comment]
Ok I have a similar problem but `zigbee2mqtt` is not really verbose about the reason:
```
...
zigbee2mqtt  | Zigbee2MQTT:info  2024-01-30 21:11:14: Connecting to MQTT server at mosquitto
zigbee2mqtt  | Zigbee2MQTT:debug 2024-01-30 21:11:14: Using MQTT anonymous login
zigbee2mqtt  | Zigbee2MQTT:debug 2024-01-30 21:11:14: Using MQTT client ID: 'zigbee2mqtt'
zigbee2mqtt  | Zigbee2MQTT:error 2024-01-30 21:11:14: MQTT failed to connect, exiting...
zigbee2mqtt  | Zigbee2MQTT:info  2024-01-30 21:11:14: Stopping zigbee-herdsman...
zigbee2mqtt  | Zigbee2MQTT:info  2024-01-30 21:11:25: Stopped zigbee-herdsman
```
with other client I can successfully connect without any issues (even with TLS, verify and credentials). Even from within the container!

**Comment by sfnemis** [comment]
> > In my case the problem with "not authorized MQTT fail" was in mosquitto configuration. Everything work after add in mosquitto brocker:
> > ```
> > loggins:
> >    - username: mqtt-user
> >      password: Anim@lsH0m3
> > ```
> 
> this sorted out my issue as well. thank you.

Thank you so much ! It worked !

**Comment by Bart9282** [comment]
> For future future travellers, essentially make your password alphanumeric only (I had a # in mine that likely borked it), and be sure to change it on both ends. Reconfigure the MQTT integration and the Zigbee2mqtt side either in settings>MQTT, or in the configuration.yaml. I was initially struggling because I forgot the MQTT integration side.

OMG after 4 hours of trying all sort of things ( I am a newbie)  this fixed it. Thank you very much!!!

**Comment by djay07** [comment]
I have a wierd behavior, same error msg, but whenever i try to start z2m it changes the user in the confi.yml to "addon"
`user: addons`

anyone else with that? - i don't really know what to do. I am trying for hours now.

Edit:
I stopped the broker, z2m, adapted the user and pw. restarted them and suddently it works........ puzzled and creating a backup.
